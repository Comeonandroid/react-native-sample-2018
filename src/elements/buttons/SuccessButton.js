import React from 'react'
import { PropTypes } from 'prop-types'

import BaseButton from './BaseButton'

const GRADIENT = ['#2a986f', '#06d947']

const SuccessButton = ({ children, rounded, size, onPress, enabled }) => (
  <BaseButton
    gradient={GRADIENT}
    rounded={rounded}
    size={size}
    onPress={onPress}
    enabled={enabled}
    >
    {children}
  </BaseButton>
)

SuccessButton.propTypes = {
  children: PropTypes.string.isRequired,
  rounded: PropTypes.bool.isRequired,
  size: PropTypes.oneOf(['small', 'large']).isRequired,
  enabled: PropTypes.bool,
  onPress: PropTypes.func
}

SuccessButton.defaultProps = {
  rounded: false,
  size: 'small'
}

export default SuccessButton