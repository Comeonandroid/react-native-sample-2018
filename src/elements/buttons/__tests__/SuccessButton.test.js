import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import SuccessButton from '../SuccessButton'

const setup = propOverrides => {
  const props = Object.assign({
    children: 'Test',
    rounded: false,
    size: 'small',
    enabled: true,
    onPress: jest.fn(),
  }, propOverrides)

  const component = <SuccessButton {...props} />
  const wrapper = shallow(component)

  return {
    props,
    component,
    wrapper
  }
}

describe('<SuccessButton />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })
})
