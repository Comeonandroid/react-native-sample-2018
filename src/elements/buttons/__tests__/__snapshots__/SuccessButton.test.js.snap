// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessButton /> dark: renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4280981615,
      4278638919,
    ]
  }
  enabled={true}
  endPoint={undefined}
  locations={null}
  onPress={[MockFunction]}
  rounded={false}
  startPoint={undefined}
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      },
      undefined,
    ]
  }
>
  <MKTouchable
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    className={undefined}
    enabled={true}
    hitSlop={undefined}
    maskBorderRadius={2}
    maskColor="rgba(255,255,255,0.15)"
    maskDuration={200}
    maskEnabled={true}
    nativeID={undefined}
    onChange={[Function]}
    onLayout={[Function]}
    onPress={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTouch={[Function]}
    pointerEvents="box-only"
    rippleColor="rgba(255,255,255,0.2)"
    rippleDuration={200}
    rippleLocation="tapLocation"
    shadowAniEnabled={true}
    shadowColor="black"
    shadowOffset={
      Object {
        "height": 0,
        "width": 0,
      }
    }
    shadowOpacity={0}
    shadowRadius={0}
    style={
      Array [
        Array [
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            },
            undefined,
          ],
          Object {},
        ],
        Object {
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      pointerEvents="none"
      size="small"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Open Sans",
            "fontSize": 16,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
    >
      Test
    </Text>
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.15)",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 0,
            "height": 0,
            "left": 0,
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
  </MKTouchable>
</BVLinearGradient>
`;

exports[`<SuccessButton /> light: renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4280981615,
      4278638919,
    ]
  }
  enabled={true}
  endPoint={undefined}
  locations={null}
  onPress={[MockFunction]}
  rounded={false}
  startPoint={undefined}
  style={
    Array [
      Object {
        "borderRadius": 0,
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "opacity": 1,
      },
      undefined,
    ]
  }
>
  <MKTouchable
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    className={undefined}
    enabled={true}
    hitSlop={undefined}
    maskBorderRadius={2}
    maskColor="rgba(255,255,255,0.15)"
    maskDuration={200}
    maskEnabled={true}
    nativeID={undefined}
    onChange={[Function]}
    onLayout={[Function]}
    onPress={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onTouch={[Function]}
    pointerEvents="box-only"
    rippleColor="rgba(255,255,255,0.2)"
    rippleDuration={200}
    rippleLocation="tapLocation"
    shadowAniEnabled={true}
    shadowColor="black"
    shadowOffset={
      Object {
        "height": 0,
        "width": 0,
      }
    }
    shadowOpacity={0}
    shadowRadius={0}
    style={
      Array [
        Array [
          Array [
            Object {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
              "opacity": 1,
            },
            undefined,
          ],
          Object {},
        ],
        Object {
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      pointerEvents="none"
      size="small"
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "Open Sans",
            "fontSize": 16,
            "fontWeight": "600",
          },
          undefined,
        ]
      }
    >
      Test
    </Text>
    <View
      collapsable={undefined}
      style={
        Object {
          "backgroundColor": "rgba(255,255,255,0.15)",
          "borderRadius": 0,
          "height": 1,
          "left": -0,
          "opacity": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": -0,
          "width": 1,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgba(255,255,255,0.2)",
            "borderRadius": 0,
            "height": 0,
            "left": 0,
            "top": 0,
            "transform": Array [
              Object {
                "scale": 0,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
  </MKTouchable>
</BVLinearGradient>
`;
